### PR TITLE
#1-4Task Andrey Yakshuk

### PR DESCRIPTION
**Pull request by andreyPromaster**

### CODE REVIEW

**Comment by FedorovD** [comment]
travis link?

**Comment by andreyPromaster** [comment]
https://travis-ci.com/github/andreyPromaster/js-assignments
Im so sorry :)

File: task/01-strings-tasks.js [hunk 1]
```diff
@@ -69,7 +70,7 @@ function getStringFromTemplate(firstName, lastName) {
  *   'Hello, Chuck Norris!' => 'Chuck Norris'
  */
 function extractNameFromTemplate(value) {
-    throw new Error('Not implemented');
+    return value.slice(7, value.length - 1)
```
Comment (FedorovD) [inline task/01-strings-tasks.js:73]: looks good, but please note slice can work with negative numbers also to take symbols from end. In the case `slice(7, -1);` can do the same.

File: task/01-strings-tasks.js [hunk 2]
```diff
@@ -201,7 +203,27 @@ function extractEmails(str) {
...
+            str += '┘\n';
+        } else
+        {
+            str += '│' ;
+            str += ' '.repeat(width - 2);
+            str += '│\n';
+        }
+    }
+    return str;
 }
```
Comment (FedorovD) [inline task/01-strings-tasks.js:227]: a lot of extra indentation, see https://javascript.info/coding-style

File: task/03-date-tasks.js [hunk 3]
```diff
@@ -56,7 +56,17 @@ function parseDataFromIso8601(value) {
...
+   {
+      return false;
+   } else if(year % 100 != 0)
+   {
+      return true;
+   } else if(year % 400 != 0)
+   {
+      return false;
+   } else return true;
 }
```
Comment (FedorovD) [inline task/03-date-tasks.js:70]: extra indentation

File: task/02-numbers-tasks.js [hunk 4]
```diff
@@ -202,7 +203,15 @@ function roundToPowerOfTen(num, pow) {
...
+    for(let i=2; i <= Math.sqrt(n); i++)
+    {
+    if(n % i==0)
+    {
+        return false;
+    }
+    }
+    return true;
+    
 }
```
Comment (FedorovD) [inline task/02-numbers-tasks.js:215]: extra indentation

**Comment by FedorovD** [comment]
Closing the PR as accepted! Thanks!

**Pull request closed without merging**